manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -514,6 +528,9 @@ func (r *FybrikApplicationReconciler) constructDataInfo(req *DataInfo, appContex
 		}
 		logging.LogStructure("Catalog connector response", response, &log, zerolog.DebugLevel, false, false)
 		response.DeepCopyInto(req.DataDetails)
+	} else if req.Context.Requirements.FlowParams.ResourceMetadata != nil {
+		// Fill req.DataDetails with the metadata from the fybrikapplication
+		req.DataDetails.ResourceMetadata = *req.Context.Requirements.FlowParams.ResourceMetadata.DeepCopy()
```

Comment:
Why is DeepCopy needed?

Reply:
@shlomitk1 to avoid changing the values when it further processed.

Reply:
`req.DataDetails.ResourceMetadata = *req.Context.Requirements.FlowParams.ResourceMetadata` is enough, the structure is copied.

Reply:
Fixed Thanks